### ISSUE TITLE
Error building dita

### ISSUE THREAD

**Issue by undavide** [grefel/extendscriptApiDocTransformations#1]
Hello Gregor,
I'm trying to borrow your project to build Photoshop output. Alas my knowledge of the tools you use are below zero (never heard before about Saxon, dita, etc), so I'm just guessing _wildly_.

So far I've been able to (I'm writing here also as a reminder for my future self, and everyone Photoshop-oriented):
- get `photoshop.xml`, `javascript.xml`, `scriptui.xml`.
- substitute all the instances of indesign with photoshop in the `mergeFiles.xslt`.
- run `java -jar saxon9he.jar -xsl:mergeFiles.xslt -it:mergeDOMFiles -o:photoshop-merged.xml`
-  run `java -jar saxon9he.jar -s:photoshop-merged.xml -xsl:fixDom.xsl -o:photoshop-merged-fixed.xml`
- run `java -jar saxon9he.jar -s:photoshop-merged-fixed.xml -xsl:dom2dita.xsl -o:domOut/*`

256 `.dita` files are generated in `/domOut`, but I get the following error:

> java -jar saxon9he.jar -s:photoshop-merged-fixed.xml -xsl:dom2dita.xsl -o:domOut/*
> Type error at char 17 in xsl:variable/@select on line 781 column 79 of dom2dita.xsl:
>   XPTY0004: A sequence of more than one item is not allowed as the first argument of
>   generate-id() (<classdef/>, <classdef/>) 
>   at xsl:apply-templates (file:/Users/davidebarranca/Downloads/extendscriptApiDocTransformations-master/dom2dita.xsl#18)
>      processing /domRoot/indd[1]/dictionary[1]/package[1]/classdef[200]
> A sequence of more than one item is not allowed as the first argument of generate-id() (<classdef/>, <classdef/>) 

Referring to `\<classdef/>` and peeking in the `photoshop-merged-fixed.xml` I assume something went wrong in the extendscript/scriptui side (not photoshop) but I cannot be sure. I can't say whether all the files have been generated or there's something missing either.

If you want to test yourself, you can download my [`photoshop-merged-fixed.xml`](https://dl.dropboxusercontent.com/u/23243188/photoshop-merged-fixed.xml.zip).

Besides, any extra guidance that you can offer from the .dita onwards (generating html output, and especially Sublime Text Code Completions) is _greatly appreciated_.

Thank you very much in advance,

Davide Barranca

**Comment by grefel** [comment]
Hey Davide, I just returned from vacation but try to look into this as soon as possible.

**Comment by grefel** [comment]
Hey Davide I fixed the photoshop issue wiht https://github.com/grefel/extendscriptApiDocTransformations/commit/ace348fabded5704fa77f421a763f17c8b4fa6c0

I tested with a CC2014 Object Model (could not find the newer one quickly) and put the results on http://www.indesignjs.de/extendscriptAPI/photoshop2014/

or as complete zip http://www.indesignjs.de/extendscriptAPI/photoshop2014.zip

The Sublime Completions are there as well

http://www.indesignjs.de/extendscriptAPI/photoshop2014/jsx-photoshop.sublime-completions

I create the webhelp output with oxygen https://www.oxygenxml.com/ which is more or less a one button solution. I've never put up a dita open toolkit by hand. 

I can offer you to transform your results and if you're interested in optimizing the html result i could also put my css and javascript extras online.

**Comment by grefel** [comment]
I put 2015.5 online

http://www.indesignjs.de/extendscriptAPI/photoshop2015.5/

**Comment by undavide** [comment]
Hi Gregor,
thanks for the great job!
I really like the doc you've put online – I'm afraid I can't produce such a nice output myself because, as far as I understand, it requires a paid version of Oxygen (and knowledge on how to use it).
I've used the commandline from http://www.dita-ot.org/ (producing HTML5 but I guess I've missing template pages and styling – for I get something like:

![image](https://cloud.githubusercontent.com/assets/5939127/18675588/1f8fefb0-7f53-11e6-858d-7af5142b60f4.png)

Anyway, not a big issue – it was more of an experiment of mine. You'll be pinged when Photoshop-next will be released :-) 

I'd like to thank you, and ask how come (if you happen to know it) is that – following all the steps – my `jsx.sublime-completions` is very different from yours.
Not that it's much shorter (~1700 lines compared to ~5500 – if you have also ID classes it makes sense that your file is bigger); but I've noticed that yours features composed strings such as `"BlendMode.COLOR"`, or `"DefaultRenderingIntent.ABSOLUTE_COLORIMETRIC"`, while mine is just `"blendMode"` and `"COLOR"`, `"DefaultRenderingIntent"` and `"ABSOLUTE_COLORIMETRIC"`. Just a curiosity.

Thank you again!
